lib/cdo/lazy.rb

```diff
@@ -0,0 +1,26 @@
+module Cdo
+  # Lazy-loads a delegated object from a block.
+  #
+  # Example:
+  #
+  # > lazy = Cdo.lazy {puts 'loaded!'; 'Lazy'}; true
+  # => true
+  # > puts lazy
+  # loaded!
+  # Lazy
+  # => nil
+  class Lazy < SimpleDelegator
```

Comment:
Even though it's implicitly covered already, it's be worth adding dedicated unit tests for this class if it's intended to be reusable throughout the code base.

Reply:
added in e746f3e5e9444fe5b6ef733ba2d97590bd6de6d2